ci/check_and_fix_format.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+set -e
+
+function fix {
+  set +e
+  ci/do_ci.sh fix_format
+  ci/do_ci.sh fix_spelling
+  ci/do_ci.sh fix_spelling_pedantic
+  echo "Format check faild, try apply following patch to fix:"
+  git diff HEAD | tee /build/fix_format.diff
```

Comment:
WDYT of allowing a command-line override of the location of the generated diff file so we can run this outside of docker?

Reply:
Added env var override, though `ci/do_ci.sh` is hard to run outside of docker anyway.